### ISSUE TITLE
Extra newline in cert files results in 'java.io.IOException: Empty input'

### ISSUE THREAD

**Issue by rfalke** [logstash-plugins/logstash-input-tcp#144]
- Version: logstash-input-tcp-5.2.2, logstash: 6.7.1
- Operating System: Linux
- Config File (if you have sensitive info, please remove it):

With an extra line in the certificate file I get
```
java.security.cert.CertificateException: Could not parse certificate: java.io.IOException: Empty input, :backtrace=>[
"sun.security.provider.X509Factory.engineGenerateCertificate(sun/security/provider/X509Factory.java:110)",
"java.security.cert.CertificateFactory.generateCertificate(java/security/cert/CertificateFactory.java:339)",
"java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:498)",
"org.jruby.javasupport.JavaMethod.invokeDirectWithExceptionHandling(org/jruby/javasupport/JavaMethod.java:455)",
"org.jruby.javasupport.JavaMethod.invokeDirect(org/jruby/javasupport/JavaMethod.java:316)",
"usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.logstash_minus_input_minus_tcp_minus_5_dot_2_dot_2_minus_java.lib.logstash.inputs.tcp.compat_ssl_options.fetch_certificates_from_file(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-input-tcp-5.2.2-java/lib/logstash/inputs/tcp/compat_ssl_options.rb:133)",
"usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.logstash_minus_input_minus_tcp_minus_5_dot_2_dot_2_minus_java.lib.logstash.inputs.tcp.compat_ssl_options.toSslContext(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-input-tcp-5.2.2-java/lib/logstash/inputs/tcp/compat_ssl_options.rb:106)",
"org.jruby.java.proxies.ArrayJavaProxy.each(org/jruby/java/proxies/ArrayJavaProxy.java:450)",
"usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.logstash_minus_input_minus_tcp_minus_5_dot_2_dot_2_minus_java.lib.logstash.inputs.tcp.compat_ssl_options.toSslContext(/usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-input-tcp-5.2.2-java/lib/logstash/inputs/tcp/compat_ssl_options.rb:105)",
"usr.share.logstash.vendor.bundle.jruby.$2_dot_5_dot_0.gems.logstash_minus_input_minus_tcp_minus_5_dot_2_dot_2_minus_java.lib.logstash.inputs.tcp.compat_ssl_options.RUBY$method$toSslContext$0$__VARARGS__(usr/share/logstash/vendor/bundle/jruby/$2_dot_5_dot_0/gems/logstash_minus_input_minus_tcp_minus_5_dot_2_dot_2_minus_java/lib/logstash/inputs/tcp//usr/share/logstash/vendor/bundle/jruby/2.5.0/gems/logstash-input-tcp-5.2.2-java/lib/logstash/inputs/tcp/compat_ssl_options.rb)",
...
```

This was working with logstash-input-tcp-5.0.9, logstash 6.3.1. The certificate files with newline are also working with the beats input of logstash 6.7.1.

**Comment by djmcgreal-cc** [comment]
Narrowed down to between 5.2.0 and 5.2.2 (regression between Logstash 6.6.2 and 6.7.0). I'm suspicious of https://github.com/logstash-plugins/logstash-input-tcp/commit/6a341cbcacfbbdf192d3f26f985f6fedd6597cad

**Comment by jsvd** [comment]
versions 5.2.3 and 6.0.3 have been shipped with the fix #145

**Comment by bangejsans** [comment]
In case anyone is having this issue with 7.X, removing any newlines from the end of my cert file resolved this same error.
